addons/spectator/XEH_postInit.sqf

```diff
@@ -0,0 +1,96 @@
+/*
+    Author:
+    voiper
+*/
+
+#include "script_component.hpp"
+
+["SettingsInitialized", {
+    if !GVAR(enabled) exitWith {};
+    
+    //check if respawn is set up properly
+    _fail = if (getNumber (missionConfigFile >> "respawn") != 3 && getText (missionConfigFile >> "respawn") != "Base") then {true} else {false};
+    if (_fail) exitWith {
+        _text = "[ACE_Spectator] ERROR: This mission does not have respawn set up properly. Add 'respawn=3' or 'respawn=""BASE""' to description.ext.";
+        systemChat _text;
+        diag_log text _text;
+    };
+    
+    if GVAR(endMission) then {
+        [{
+            if (player distance GVAR(penPos) < 200) then {
+                if ({isPlayer _x && alive _x && (_x distance GVAR(penPos)) > 200} count allUnits == 0) then {
+                    [["endDeath", false], "BIS_fnc_endMission"] call BIS_fnc_MP;
+                    [_this select 1] call CBA_fnc_removePerFrameHandler;
+                };
+            };
+        }, 2] call CBA_fnc_addPerFrameHandler;
+    };
+    
+    if (isDedicated) exitWith {};
+    
+    call FUNC(penPos);
+
+    {
+        if (getMarkerPos _x isEqualTo [0,0,0]) then {
```

Comment:
Assuming this code will stay the same (I have some ideas for future iterations) it should probably become `!(_x in allMapMarkers)`